core/src/main/java/jenkins/util/VirtualFile.java

```diff
@@ -119,13 +170,75 @@
      */
     public abstract @Nonnull VirtualFile[] list() throws IOException;
 
+    /**
+     * @deprecated use {@link #list(String, String, boolean)} instead
+     */
+    @Deprecated
```

Comment:
This deprecation points to a `Beta` API. This seems unhelpful.

Reply:
Well, the only callers will be updated anyway.